osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/deploymentspec/OsSvaServerCreateTask.java

```diff
@@ -160,24 +161,34 @@ public void executeTransaction(EntityManager em) throws Exception {
         this.activeRunner.getOsDeploymentSpecNotificationRunner()
                 .addSVAIdToListener(this.dai.getDeploymentSpec().getId(), createdServer.getServerId());
 
-            if (vc.isControllerDefined()) {
-                try {
-                    DefaultNetworkPort ingressPort = new DefaultNetworkPort(createdServer.getIngressInspectionPortId(),
-                            createdServer.getIngressInspectionMacAddr());
-                    DefaultNetworkPort egressPort = new DefaultNetworkPort(createdServer.getEgressInspectionPortId(),
-                            createdServer.getEgressInspectionMacAddr());
-
-                    if (this.apiFactoryService.supportsPortGroup(this.dai.getVirtualSystem())) {
-                        String domainId = OpenstackUtil.extractDomainId(
-                                ds.getProjectId(),
-                                ds.getProjectName(),
-                                ds.getVirtualSystem().getVirtualizationConnector(),
-                                Arrays.asList(ingressPort));
-                        ingressPort.setParentId(domainId);
-                        egressPort.setParentId(domainId);
-                    }
-                    //Element object in DefaultInspectionport is not used at this point, hence null
-                    controller.registerInspectionPort(new DefaultInspectionPort(ingressPort, egressPort, null));
+        if (vc.isControllerDefined()) {
+            try {
+                DefaultNetworkPort ingressPort = new DefaultNetworkPort(createdServer.getIngressInspectionPortId(),
+                        createdServer.getIngressInspectionMacAddr());
+                DefaultNetworkPort egressPort = new DefaultNetworkPort(createdServer.getEgressInspectionPortId(),
+                        createdServer.getEgressInspectionMacAddr());
+
+                String redirectionTargetId = null;
+
+                if (this.apiFactoryService.supportsPortGroup(this.dai.getVirtualSystem())) {
+                    String domainId = OpenstackUtil.extractDomainId(
+                            ds.getProjectId(),
+                            ds.getProjectName(),
+                            ds.getVirtualSystem().getVirtualizationConnector(),
+                            Arrays.asList(ingressPort));
+                    ingressPort.setParentId(domainId);
+                    egressPort.setParentId(domainId);
+
+                    redirectionTargetId = ds.getRedirectionTargetId();
+                }
+
+                //Element object in DefaultInspectionport is not used at this point, hence null
+                Element element = controller.registerInspectionPort(new DefaultInspectionPort(ingressPort, egressPort,
+                        null, redirectionTargetId));
+
+                    // TODO : at least one sdn plugin will result in element.getParentId being non-null, whether or
```

Comment:
it should be fine to leave this as is. the redirection target id will be set to the right value for any plugin.